flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java

```diff
@@ -284,10 +293,58 @@ private static String toHtmlString(final OptionWithMetaInfo optionWithMetaInfo)
 			"        <tr>\n" +
 			"            <td><h5>" + escapeCharacters(option.key()) + "</h5>" + execModeStringBuilder.toString() + "</td>\n" +
 			"            <td style=\"word-wrap: break-word;\">" + escapeCharacters(addWordBreakOpportunities(defaultValue)) + "</td>\n" +
+			"            <td>" + type + "</td>\n" +
 			"            <td>" + formatter.format(option.description()) + "</td>\n" +
 			"        </tr>\n";
 	}
 
+	private static Class<?> getClazz(ConfigOption<?> option) {
+		try {
+			Method getClazzMethod = ConfigOption.class.getDeclaredMethod("getClazz");
+			getClazzMethod.setAccessible(true);
+			Class clazz = (Class) getClazzMethod.invoke(option);
+			getClazzMethod.setAccessible(false);
+			return clazz;
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	private static boolean isList(ConfigOption<?> option) {
+		try {
+			Method getClazzMethod = ConfigOption.class.getDeclaredMethod("isList");
+			getClazzMethod.setAccessible(true);
+			boolean isList = (boolean) getClazzMethod.invoke(option);
+			getClazzMethod.setAccessible(false);
+			return isList;
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@VisibleForTesting
+	static String typeToHtml(OptionWithMetaInfo optionWithMetaInfo) {
+		ConfigOption<?> option = optionWithMetaInfo.option;
+		Class<?> clazz = getClazz(option);
+
```

Comment:
Do we expect to have a list of enums or of maps (i.e. any level of "nesting")? If yes, what do we expect to have as the result for the docs here?

Reply:
That's a good point I forgot about. We do support a single level of nesting. So there might be a List<Map> or List<Enum>. I improved the doc generator to handle those cases as well.